### PR TITLE
Initial support for Azure Tickets tap

### PR DESCRIPTION
**Pull request by nsmith22**
Initial support for Azure Tickets tap.

Take note that all the code is already in `main` and this PR is created in the reverse of normal direction. It will not be merged, but is created to facilitate review.

# Testing
Created an Azure PAT and added it to the DB using the recently developed `integrations` page UI. Then, ran the tap locally via the `scheduled` repo against the minware Azure DevOps org at https://dev.azure.com/minware/.

* Ran with `target = file` and verified correctness of the resulting output file to ensure it contained the expected projects, boards, teams, work items, etc.
* Ran with `target = snowflake` (pointed at dev env) and verified correctness of the resulting table schemas and data. Also verified correct state data was written to S3.
* Ran the same as above and verified that no new work items or updates were ingested because no changes had been made since the previous run (as tracked via Singer bookmark state).
* Modified a single ticket, ran the same as above, and verified that exactly one work item (the modified one) was ingested again as well as the updates for it.

# Data
The resulting data can be inspected in the schema `MINWARE_DEV.T_00000000000000000000000000000011_AZURETICKETS`.

### CODE REVIEW

File: tap_azure_tickets/__init__.py [hunk 1]
```diff
@@ -0,0 +1,779 @@
...
+logger = singer.get_logger()
+
+# globals which hold data across streams
+workItemIds = []
+
+# map of Azure field names to more usable/readable field names
+FIELD_MAP = {
+    "System.Id": "id",
+    "System.AssignedTo": "assignedTo",
+    "Microsoft.VSTS.Common.ClosedBy": "closedBy",
```
Comment (KBorders01) [inline tap_azure_tickets/__init__.py:26]: Gotta love Microsoft, this reminds me of IE 6, lol.

**Review by KBorders01 — APPROVED** [review]

**Comment by KBorders01** [comment]
Looks good @nsmith22, don't think I have any feedback to add!

File: tap_azure_tickets/__init__.py (hunk 1 above)
Comment (nsmith22) [reply tap_azure_tickets/__init__.py:26]: Yeah if it weren't for the non-engineers at that company, their main website would be probably be as bad as something like https://vsts.common.public.marketing.microsoft.com/js-version=7.0-preview&$compat=ie

**Comment by KBorders01** [comment]
Haha

**Comment by nsmith22** [comment]
Closing since all the code is already in `main` as mentioned in the PR description. Nothing to merge, and this PR was opened against a branch for the purposes of reviewing the diff.

**Pull request closed without merging**